### PR TITLE
[analyzer] Fix nullptr dereference for symbols from pointer invalidation

### PR DESCRIPTION
**Pull request by necto**
As reported in https://github.com/llvm/llvm-project/pull/105648#issuecomment-2317144635 commit 08ad8dc7154bf3ab79f750e6d5fb7df597c7601a
introduced a nullptr dereference in the case when store contains a binding to a symbol that has no origin region associated with it, such as the symbol generated when a pointer is passed to an opaque function.

### CODE REVIEW

**Review by Xazax-hun — APPROVED** [review]
LGTM!

File: clang/test/Analysis/stack-addr-ps.c [hunk 1]
```diff
@@ -126,3 +126,22 @@ void caller_for_nested_leaking() {
   int *ptr = 0;
   caller_mid_for_nested_leaking(&ptr);
 }
+
+// This used to crash StackAddrEscapeChecker because
+// it features a symbol conj_$1{struct c *, LC1, S763, #1}
+// that has no origin region.
+// bbi-98571
```
Comment (steakhal) [inline clang/test/Analysis/stack-addr-ps.c:133]: Tests are not supposed to have references to tickets that are not publicly reachable. Swap it for a link if you really want to preserve this.

File: clang/test/Analysis/stack-addr-ps.c [hunk 2]
```diff
@@ -126,3 +126,22 @@ void caller_for_nested_leaking() {
...
+struct a {
+  int member;
+};
+
+struct c {
+  struct a *nested_ptr;
+};
+long global_var;
+void opaque(struct c*);
+void bbi_98571_no_crash() {
```
Comment (steakhal) [inline clang/test/Analysis/stack-addr-ps.c:143]: I'd recommend using a different name, without the bbi part.

File: clang/test/Analysis/stack-addr-ps.c [hunk 3]
```diff
@@ -126,3 +126,22 @@ void caller_for_nested_leaking() {
...
+  struct c *ptr = (struct c *)global_var;
```
Comment (steakhal) [inline clang/test/Analysis/stack-addr-ps.c:144]: Isn't this cast platform dependent?
If so, pin the target.

File: clang/test/Analysis/stack-addr-ps.c [hunk 4]
```diff
@@ -126,3 +126,22 @@ void caller_for_nested_leaking() {
...
+  opaque(ptr);
+  ptr->nested_ptr->member++;
```
Comment (steakhal) [inline clang/test/Analysis/stack-addr-ps.c:146]: Did it crash at this line? If so, mark it with zhe no-crash comment.

**Review by steakhal — CHANGES_REQUESTED** [review]

File: clang/test/Analysis/stack-addr-ps.c (hunk 1 above)
Comment (necto) [reply clang/test/Analysis/stack-addr-ps.c]: removed in 7cd8719

File: clang/test/Analysis/stack-addr-ps.c (hunk 2 above)
Comment (necto) [reply clang/test/Analysis/stack-addr-ps.c]: renamed in 7cd8719

File: clang/test/Analysis/stack-addr-ps.c (hunk 4 above)
Comment (necto) [reply clang/test/Analysis/stack-addr-ps.c:145]: noted in 
d78657a

**Review by steakhal — APPROVED** [review]

**Pull request merged**